networks/known_networks.go

```diff
@@ -771,30 +776,48 @@ func MustSetNetworks(networkCfg config.NetworkConfig) []blockchain.EVMNetwork {
 				panic(fmt.Errorf("no wallet keys found in config for '%s' network", selectedNetworks[i]))
 			}
 		}
-		network, err := NewEVMNetwork(networkName, walletKeys, httpUrls, wsUrls)
-		if err != nil {
-			panic(err)
+		// if evm_network config is found, use it
+		if networkCfg.EVMNetworks != nil {
+			if network, ok := networkCfg.EVMNetworks[networkName]; ok && network != nil {
+				if err := NewEVMNetwork(network, walletKeys, httpUrls, wsUrls); err != nil {
```

Comment:
I think we might have a problem here... consider this config:
```
[Network]
selected_networks=["polygon_mumbai"]

[Network.RpcHttpUrls]
polygon_mumbai = ["https://polygon-endpoint.io"]

[Network.RpcWsUrls]
polygon_mumbai = ["wss://polygon-endpoint.io"]

[Network.WalletKeys]
polygon_mumbai = ["change-me-to-your-PK"]

[Network.EVMNetwork.polygon_mumbai]
evm_urls = ["some-other-url"]
yet_another_onet=["evm_http_urls"]
evm_keys=["some-keys"]
```

Which one will be used? We will create instance of `NewEVMNetwork` using `Netowrk.RpcHttpUrls`, but then use the RPCs defined under `EVMNetwork`.

No?

Reply:
It will still use `[Network.RpcHttpUrls]` `[Network.RpcWsUrls]` and` [Network.WalletKeys]`
Let me add a test for this scenario

Reply:
Is that what we want? How do we make it clear to the user that RPCs passed under `EVMNetwork` config will be ignored? Throw error when both are set?